### ISSUE TITLE
Datepicker: Initialized date in ngModel is not displayed in input field

### ISSUE THREAD

**Issue by cleptomaniac361** [ng-bootstrap/ng-bootstrap#2279]
### Bug description:
Regarding 'Select Today' [@Custom date adapter example](https://ng-bootstrap.github.io/#/components/datepicker/examples).  
It is not possible to add a predefined date to the data picker from outside via ngModel. The date is stored correctly in the component, but the date is not displayed in the input field of the datepicker. If the model changes, e.g. if the button is activated, the change is also not visible on the surface.

### Link to minimally-working plunker that reproduces the issue:
[http://plnkr.co/edit/OKerlvUnWAD7gvIC9hWv?p=preview](http://plnkr.co/edit/OKerlvUnWAD7gvIC9hWv?p=preview)

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 5.2.9

ng-bootstrap: 1.1.0

Bootstrap: 4.0.0

**Comment by pkozlowski-opensource** [comment]
I'm confused - in your example you are _not_ using a custom adapter... When you actually _use_ it things seem to work correctly: http://plnkr.co/edit/gwKmQyh4JZQjhYzjmM54?p=preview

**Comment by cleptomaniac361** [comment]
Okay, the mention of the adapter is admittedly confusing. To make it even clearer: my main concern is to pass an initialized date to the data picker component. When the component is displayed for the first time, a date should already be visible in the input field of the component. In my example, today's date. As you can see in my Plunkr, the Input field remains empty or the placeholder is displayed.

In summary: It is not about the adapter but about how I can transfer predefined data to the data picker.

Translated with www.DeepL.com/Translator

**Comment by pkozlowski-opensource** [comment]
To properly set initial model value you need to use date _type_ expected by the datepicker component. By default it is [NgbDateStruct](https://ng-bootstrap.github.io/#/components/datepicker/api#NgbDateStruct) and you need to use it if you don't need to use any date type conversion adapters: http://plnkr.co/edit/1aYx748WdFiux4jsb8a8?p=preview

If you don't want to use `NgbDateStruct` but JS `Date` objects instead then you _must_ use a date type conversion adapter and the plunker I've linked before illustrates how to do it: http://plnkr.co/edit/gwKmQyh4JZQjhYzjmM54?p=preview

In short: date object _type_ must much type expected by the datepicker. You can configure datepicker to work with different date types through adapters.

Closing as a support request.

**Comment by leerenold** [comment]
Initialized date doesn't display the correct month when using ngbDatepicker as component ,though it sets the date correctly as per model. (
Expected behavior: View should switch to month set using model value.
It works correctly when using ngbDatepicker as directive.
